summarization/summarizer.py

```diff
@@ -377,3 +381,21 @@ def _get_stamp_descriptor_for_embedded_content(self, content):
         # type casting is required since the dict keys
         # are now of type str
         return self.embedded_descriptors_dict[str(content.content_type)]
+
+    def _collect_embedded_and_quoted_indices(self):
+        self.embedded_indices = set([
+            content.content_index for content in self.contents.embedded_content
+        ])
+
+        self.quoted_indices = set([
+            content.content_index for content in self.contents.quoted_content
+        ])
+
+    def _set_stamp_page_types(self):
+        for stamp_page in self.stamp_pages_list:
+            stamp_page.set_stamp_type(
+                stamp_page.get_stamp_page_type(
+                    self.embedded_indices,
+                    self.quoted_indices
+                )
```

Comment:
`stamp_page.update_stamp_page_type(self.embedded_indices, self.quoted_indices)
`

Reply:
Defer this method inside method to StampPage

Reply:
updated :D